File: Dynamic_Programming/377.Combination-Sum-IV/377.Combination-Sum-IV.py
class Solution:
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        dp = [1]+[0]*target
        for i in range(1,target+1):
            for j in range(len(nums)):
                if i-nums[j]>=0:
                    dp[i]+=dp[i-nums[j]]
        return dp[target]
        
                

File: Dynamic_Programming/377.Combination-Sum-IV/377.Combination-Sum-IV.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) 
    {
        vector<unsigned int>dp(target+1,0);
        dp[0] = 1;
        for (int sum=1; sum<=target; sum++)
        {
            for (int x: nums)
            {
                if (sum-x>=0)
                    dp[sum] += dp[sum-x];
            }
        }
        return dp[target];
    }
};
